react-search-component/my-app/src/Search.tsx

```diff
@@ -0,0 +1,30 @@
+import { useState } from 'react';
+import { List } from './List';
+import { SearchBar } from './SearchBar';
+
+type Props = {
+  quotes: string[];
+};
+
+export function Search({ quotes }: Props) {
+  const [value, setValue] = useState('');
+
+  function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
+    setValue(e.target.value);
+  }
+
+  let filteredQuotes = quotes.filter((quote) =>
+    quote.toLowerCase().includes(value.toLowerCase())
+  );
+
+  if (filteredQuotes.length === 0) {
+    filteredQuotes = ['There are no quotes matching your search'];
```

Comment:
This is a clever way to do this, but the problem is that it isn't clear to the user that `"There are no quotes matching your search"` isn't one of the quotes!

The best way to do it is to render just that string and _not_ the List. That way you can style it so it's clear there are no matching quotes.

You an do that easily with conditional rendering:
```
{ filteredQuotes.length === 0 ?
    <div>There are no quotes matching your search</div> :
    <List list={filteredQuotes} />}
```